### ISSUE TITLE
camera conventions

### ISSUE THREAD

**Issue by SeanChen0220** [NVlabs/eg3d#65]
Hi, is the world coordinate right? It seems strange.
![image](https://user-images.githubusercontent.com/53289490/188642897-51ba0844-dcf0-4647-9cf1-4a0d571e33c4.png)

![QQ截图20220906211323](https://user-images.githubusercontent.com/53289490/188644488-19bb35f4-8d6a-4a1a-9fe2-a9a83cc4e0d3.jpg)

I also check the dataset.json for more details, for example, 00023.png. 

["00023.png", [0.9999417662620544, 0.010793998837471008, 0.00011557899415493011, 0.0015220991844047327, 0.010792133398354053, -0.9998840093612671, 0.01074833795428276, -0.012054688543150803, 0.0002315831370651722, -0.010746465064585209, -0.9999422430992126, 2.699972660546436, 0.0, 0.0, 0.0, 1.0, 4.2647, 0.0, 0.5, 0.0, 4.2647, 0.5, 0.0, 0.0, 1.0]]

According to the labels, I get the euler angles for rotation matrix [-179, 0, 0] (in degrees, x,y,z) and t=[0, 0, 2.67]. So this means we only have to rotate camera coordinate around Xc axis for 180 degrees and move it along the Zc axis. However, in this way, the world coordinate is not the same as that in your shown picture.

Is the world coordinate in your shown picture not correct? Or my caculations are wrong? I hope you can reply for my questions, that will help me a lot! Thanks!

**Comment by MartinMao1101** [comment]
hi there, i also found that the 2 coordinates are using right hand and left hand respectively. I am also quite curious about this issue. Also, i am wondering how is the world coordinate defined in this project exactly. Have you by any chance found some information about your question?

**Comment by RichardChen20** [comment]
@MartinMao1101 
Yeah, I have figured out this question by myself. The Extrinsics in labels is used to transform a point in camera coordinate to the world coordinate (face).

![image](https://user-images.githubusercontent.com/53289490/215254481-246759c6-ffc4-4aac-b324-27c572529fbd.png)
Here are the coordinates.

If u have questions, you can send me a msg.

**Comment by MartinMao1101** [comment]
@SeanChen0220 
thanks quite a lot for the tips and the nice pic! I have been wondering about the world coordinate for some weeks, you really solved my problem!

**Comment by chenweikai** [comment]
@RichardChen20 May I know where did you find the dataset.json file? Is it officially released by the authors? Thanks!

**Comment by Blue-FishER** [comment]
> @RichardChen20 May I know where did you find the dataset.json file? Is it officially released by the authors? Thanks!

Take FFHQ for example, in the dataset_processing/ffhq/runme.py, search for "Downloading poses...", you will see the link.

> print("Downloading poses...")
> gdown.download('https://drive.google.com/uc?id=14mzYD1DxUjh7BGgeWKgXtLHWwvr-he1Z', 'final_crops/dataset.json', quiet=False)

**Comment by ZhenglinZhou** [comment]
Hi all,

Recently, I try to render the image from EG3D using another camera conventions code ([threestudio](https://github.com/threestudio-project/threestudio/blob/main/threestudio/data/uncond.py#L377-L399)). And I succeeded in rendering image, when I use a c2w matrix `(x_c, y_c, z_c)` with `(right, up, -lookat)`.

**Comment by RichardChen20** [comment]
> Hi all,
> 
> Recently, I try to render the image from EG3D using another camera conventions code ([threestudio](https://github.com/threestudio-project/threestudio/blob/main/threestudio/data/uncond.py#L377-L399)). And I succeeded in rendering image, when I use a c2w matrix `(x_c, y_c, z_c)` with `(right, up, -lookat)`.

Yeah, this also works. So maybe the authors put the wrong pic here. By the way, c2w matrix `(x_c, y_c, z_c)` with `(right, up, -lookat)` is the common way we use. I was really confused about the pic in the rep as the face coord is using the left hand coord .

**Comment by XinyaChen21** [comment]
Is the c2w matrix (x_c, y_c, z_c) of Shapenet car also (right, up, -lookat)? The rendered result seems wrong when set (x_c, y_c, z_c) be (right, up, -lookat) for Shapenet car.

**Comment by povolann** [comment]
> @MartinMao1101 Yeah, I have figured out this question by myself. The Extrinsics in labels is used to transform a point in camera coordinate to the world coordinate (face).
> 
> ![image](https://user-images.githubusercontent.com/53289490/215254481-246759c6-ffc4-4aac-b324-27c572529fbd.png) Here are the coordinates.
> 
> If u have questions, you can send me a msg.

Can you please explain a bit more, I am trying to use the Shapenet dataset for rendering, but I am really confused about the geometry. So, if I have the pose matrix, how exactly can I get the correct Extrinsics for EG3D?
